common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -39,25 +39,47 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
-  def cleanAmpYouTube(document: Document) = {
-
-    document.getElementsByClass("element-video").filter { element: Element =>
-      element.getElementsByTag("iframe").length != 0
-    }.foreach { element: Element =>
-      element.getElementsByTag("iframe").map { element: Element =>
-        element.remove()
+  object AmpExternalVideo {
+    def getElementTypeByUrl(videoUrl: String) : String = {
+      val youtubePattern = ".*youtube.com.*".r
+      val vimeoPattern = ".*vimeo.com.*".r
+      videoUrl match {
+        case youtubePattern() => "amp-youtube"
+        case vimeoPattern() => "amp-vimeo"
+        case _ => ""
       }
-      val youtubeUrl = element.attr("data-canonical-url")
-      youtubeUrl.split("v=").lastOption.map { youtubeId =>
-        val youtube = document.createElement("amp-youtube")
-        youtube.attr("data-videoid", youtubeId)
-        youtube.attr("width", "5")
-        youtube.attr("height", "3")
-        youtube.attr("layout", "responsive")
-        element.appendChild(youtube)
+    }
+
+    def createElement(document: Document, videoId: String, elementType: String): Element = {
+      val video = document.createElement(elementType)
+      video.attr("data-videoid", videoId)
+      video.attr("width", "5")
+      video.attr("height", "3")
+      video.attr("layout", "responsive")
+    }
+
+    def getVideoIdFromUrl(videoUrl: String): Option[String] = {
+      val youtubePattern = ".*youtube.com/watch\\?v=(.+).*".r
+      val vimeoPattern = ".*vimeo.com/(\\d+).*".r
+      videoUrl match {
+        case youtubePattern(videoId) => Some(videoId)
+        case vimeoPattern(videoId) => Some(videoId)
+        case _ => None
```

Comment:
Can we combine this with the `getElementTypeByUrl` regex?  I think we should be able to do one regex and return something that encapsulates both the element name and the video ID.